code/datums/status_effects/debuffs.dm

```diff
@@ -530,3 +530,228 @@
+	particle_holder = new(debuff_owner, /particles/melting_status)
+	particle_holder.particles.spawning = 1 + round(stacks / 2)
+
+/datum/status_effect/stacking/melting/on_remove()
+	debuff_owner = null
+	QDEL_NULL(particle_holder)
+	return ..()
+
+/datum/status_effect/stacking/melting/tick()
+	. = ..()
+	if(!debuff_owner)
+		return
+
+	playsound(debuff_owner.loc, "sound/bullets/acid_impact1.ogg", 4)
+	particle_holder.particles.spawning = 1 + round(stacks / 2)
+
+	debuff_owner.apply_damage(STATUS_EFFECT_MELTING_DAMAGE, BURN, null, FIRE)
+
+	if(!isxeno(debuff_owner))
+		return
+	var/mob/living/carbon/xenomorph/xenomorph_owner = debuff_owner
+	xenomorph_owner.adjust_sunder(STATUS_EFFECT_MELTING_SUNDER_DAMAGE)
+
+/atom/movable/screen/alert/status_effect/melting
+	name = "Melting"
+	desc = "You are melting away!"
+	icon_state = "melting"
+
+/particles/melting_status
+	icon = 'icons/effects/particles/generic_particles.dmi'
+	icon_state = "drip"
+	width = 100
+	height = 100
+	count = 1000
+	spawning = 4
+	lifespan = 10
+	fade = 8
+	velocity = list(0, 0)
+	position = generator(GEN_SPHERE, 16, 16, NORMAL_RAND)
+	drift = generator(GEN_VECTOR, list(-0.1, 0), list(0.1, 0))
+	gravity = list(0, -0.4)
+	scale = generator(GEN_VECTOR, list(0.3, 0.3), list(1, 1), NORMAL_RAND)
+	friction = -0.05
+	color = "#cc5200"
+
+// ***************************************
+// *********** Microwave
+// ***************************************
+///amount of damage done per tick by the microwave status effect
+#define MICROWAVE_STATUS_DAMAGE_MULT 5
+///duration of the microwave effect. Refreshed on application
+#define MICROWAVE_STATUS_DURATION 10 SECONDS
+
+/datum/status_effect/stacking/microwave
+	id = "microwaved"
+	tick_interval = 1 SECONDS
+	stacks = 1
+	max_stacks = null
+	stack_decay = 0
+	consumed_on_threshold = FALSE
+	alert_type = /atom/movable/screen/alert/status_effect/microwave
+	///Owner of the debuff is limited to carbons.
+	var/mob/living/carbon/debuff_owner
+	///Used for particles. Holds the particles instead of the mob. See particle_holder for documentation.
+	var/obj/effect/abstract/particle_holder/particle_holder
+
+/datum/status_effect/stacking/microwave/can_gain_stacks()
+	if(owner.status_flags & GODMODE)
+		return FALSE
+	return ..()
+
+/datum/status_effect/stacking/microwave/on_creation(mob/living/new_owner, stacks_to_apply)
+	if(new_owner.status_flags & GODMODE)
+		qdel(src)
+		return
+	. = ..()
+	debuff_owner = new_owner
+	debuff_owner.balloon_alert(debuff_owner, "microwaved!")
+	playsound(debuff_owner.loc, "sound/bullets/acid_impact1.ogg", 30)
+	particle_holder = new(debuff_owner, /particles/microwave_status)
+	particle_holder.particles.spawning = stacks * 5
+
+/datum/status_effect/stacking/microwave/on_remove()
+	debuff_owner = null
+	QDEL_NULL(particle_holder)
+	return ..()
+
+/datum/status_effect/stacking/microwave/add_stacks(stacks_added)
+	. = ..()
+	if(stacks > 3)
+		TIMER_COOLDOWN_START(src, COOLDOWN_MICROWAVE_STATUS, MICROWAVE_STATUS_DURATION)
```

Comment:
This and the timer_cooldown_check below is absolutely not what you want to be doing here. T_C_C is true if the cooldown is currently active, so as coded it'll disappear immediately once you hit a fourth stack and otherwise last forever.

Reply:
meant to be ! on the check. Other than that do you see an issue? David and MJP wanted fairly specific behavior for this, i.e. it refreshes ONLY if the stacks exceed 3, while under 3 increases the intensity.

Reply:
The only time the timer gets created is at stacks above 3, so below that it'll last forever.

Reply:
yeah added